shared/js/media/crop_resize_rotate.js

```diff
@@ -488,8 +489,49 @@ function cropResizeRotate(blob, cropRegion, outputSize, outputType,
         URL.revokeObjectURL(baseURL);
       }
 
+      var inputBlob = blob; // XXX just use a different parameter name
+
       // Finally, encode the image into a blob
-      canvas.toBlob(gotEncodedBlob, outputType || JPEG);
+
+      var postCallback = (!outputType || outputType == JPEGType) ?
+          copyMetadata : gotEncodedBlob;
+
+      canvas.toBlob(postCallback, outputType || JPEGType);
+
+      function copyMetadata(outputBlob) {
+        try {
+          LazyLoader.load(['shared/js/media/jpeg-exif.js'], () => {
+            JPEG.readExifMetaData(inputBlob, (error, metaData) => {
+
+              if (error || !metaData) {
+                gotEncodedBlob(outputBlob);
+                return;
+              }
+              metaData.Orientation = 1;
```

Comment:
Does Diego's code define symbolic constants for these property names? Can you do `metaData[JPEG.EXIF.ORIENTATION] = 1` instead?  That might be a bit cleaner, but maybe just too verbose to be worthwhile.